UofS-Pulse-Binfo/uofspb_germplasm
# 14 new population importer

## Metadata

 - Issue #14 New Population Importer (e.g. RIL, NAM) 




Documentation:

## Description



## Dependencies



## Testing?




TO TEST:
1. Pull changes to this repository and point to the branch 14-new-population-importer
2. Set permission to importer by accessing admin/people/permissions and locate Tripal in the list of modules on the left hand side panel. Scroll to the bottom to find Importer: Use the Germplasm Population Importer and check checkbox corresponding to Administrator user then click Save Permission Button.
3. Clear Cache (2x)
4. Under Tripal/Data Loaders - Access Germplasm Population Importer.
5. Since in a KP clone this modules had been installed prior, Click on Configure module to set Relationship Verb CV and default Prefix value.  I use **stock_relationship** for verb cv and **GERM** for prefix value.
6. Go back to the Importer page.
7. Create a tsv file as described by the Importer File requirements. An example file below can be used directly or as a model.

[import me.zip](https://github.com/UofS-Pulse-Binfo/uofspb_germplasm/files/10806454/import.me.zip)

You may test validation by altering values in this data file such as:
A. Re-uploading the same file will trigger germplasm already exists
B. Use a non-existing type or organism will trigger invalid type or organism, respectively
C. A completely empty file
D. Unexpected or missing column header

The file should create 3 stocks with one using a custom unquename while the other 2 will rely on the default prefix.

8. This importer is equipped with a population viewer to verify the imported stocks.
admin/tripal/extension/germplasm_population_importer/configure
and expand View Population.
